subsys/net/lib/wifi_credentials/wifi_credentials.c

```diff
@@ -20,6 +21,32 @@ K_MUTEX_DEFINE(wifi_credentials_mutex);
 static char ssid_cache[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES][WIFI_SSID_MAX_LEN];
 static size_t ssid_cache_lengths[CONFIG_WIFI_CREDENTIALS_MAX_ENTRIES];
 
+int wifi_credentials_load_static(struct wifi_connect_req_params *params)
+{
+#ifdef CONFIG_WIFI_CREDENTIALS_STATIC
```

Comment:
Please change the `#ifdef`s to use the [IS_ENABLED](https://docs.zephyrproject.org/apidoc/latest/group__sys-util.html#ga111fe4e9d63758262fc6810a782cb32a) macro to get cleaner code and better error messages.

Reply:
@gregersrygg 
Modified code to use ``if IS_ENABLED(CONFIG_WIFI_CREDENTIALS_STATIC)`` instead of 
``if (IS_ENABLED(CONFIG_WIFI_CREDENTIALS_STATIC))``
This change is to fix the compilation errors for applications that do not enable 
 ``CONFIG_WIFI_CREDENTIALS_STATIC`` option.

Reply:
Did you get compilation errors? Could it be another reason? The `IS_ENABLED` macro is supposed to handle the case when the value is not defined at all: https://docs.zephyrproject.org/apidoc/latest/group__sys-util.html#ga111fe4e9d63758262fc6810a782cb32a

Using an `#if` macro is actually discouraged in the Zephyr docs because it doesn't give any benefits vs plain `#if defined()`

Reply:
Yes, compilation errors were seen in the not defined case , as he `` CONFIG_WIFI_CREDENTIALS_STATIC_SSID`` and ``CONFIG_WIFI_CREDENTIALS_STATIC_PSK`` are available only if ``CONFIG_WIFI_CREDENTIALS_STATIC`` option is defined.
If ``CONFIG_WIFI_CREDENTIALS_STATIC`` option is not enabled, C compiler is still trying to compile the 
if (IS_ENBALED(config)) block and then the errors were seen.